model/user/user.go

```diff
@@ -305,7 +305,7 @@ func (u *DBUser) AddRole(role string) error {
 		return nil
 	}
 	update := bson.M{
-		"$push": bson.M{RolesKey: role},
+		"$addToSet": bson.M{RolesKey: role},
```

Comment:
why is $addToSet better than $push over here?

Reply:
Good catch: it's actually the same because we check if the role already exists at 304, but theoretically local data could become inconsistent. $push will always add the item to the list, while $addToSet doesn't allow duplicates, and we want to avoid duplicates, so this felt safer (and more consistent with the other auth updates).